ddtrace/contrib/jinja2/patch.py

```diff
@@ -0,0 +1,80 @@
+import jinja2
+from wrapt import wrap_function_wrapper as _w
+
+from ddtrace import tracer
+
+from ...ext import http
+from ...utils.wrappers import unwrap as _u
+
+DEFAULT_TEMPLATE_NAME = "<template>"
+
+
+def patch():
+    if getattr(jinja2, "__datadog_patch", False):
+        # already patched
+        return
+    setattr(jinja2, "__datadog_patch", True)
+    _w(jinja2, "environment.Template.render", _wrap_render)
+    _w(jinja2, "environment.Template.generate", _wrap_generate)
+    _w(jinja2, "environment.Environment.compile", _wrap_compile)
+    _w(jinja2, "environment.Environment._load_template", _wrap_load_template)
+
+
+def unpatch():
+    if not getattr(jinja2, "__datadog_patch", False):
+        return
+    setattr(jinja2, "__datadog_patch", False)
+    _u(jinja2.Template, "render")
+    _u(jinja2.Template, "generate")
+    _u(jinja2.Environment, "compile")
+    _u(jinja2.Environment, "_load_template")
+
+
+def _wrap_generate(func, instance, args, kwargs):
+    template_name = instance.name or DEFAULT_TEMPLATE_NAME
```

Comment:
It looks like `jinja2` does `'memory:%x' % id(instance)` when an `instance.name` is not set.

https://github.com/pallets/jinja/blob/7a6704db55fcd9bbe5a90c3868e03f5a5fcf176a/jinja2/environment.py#L1132-L1137

Reply:
I've seen that, I just wanted to have a common "name" between the operations. But I could use something more explicit like `<memory>` when `name is None`. I don't know if adding `id(template)` is useful.

Reply:
yeah, I don't think it is that useful either since it isn't a hash of the string, it is going to be the in-memory object id...

I think `<memory>` is great on it's own. :+1: